[PLATFORM-60] Sortable lists

mondoreale opened the pull request:
I used [`react-sortable-hoc`](https://github.com/clauderic/react-sortable-hoc) to implement 2 sortable list components:
- `SortableList` – base for anything sortable, and
- `FieldList` – more complex list where items have designated handles, and can carry pretty much any content (I used `TextInput` in my example).

![nov-20-2018 14-05-42](https://user-images.githubusercontent.com/320066/48775466-6b73c480-eccd-11e8-903f-f93667df75b5.gif)

Both available in Storybook today. Pre-order starts now! 😉

timoxley commented on app/src/shared/components/FieldList/FieldItem/index.jsx:
```diff
@@ -0,0 +1,24 @@
+    <div className={styles.root}>
+        <div className={styles.inner}>
+            <Handle className={styles.handle} />
+            <TextInput label="" defaultValue={name} />
```
How does one listen for changes on the `TextInput`? I'm thinking perhaps `TextInput` should be `{children}`?

mondoreale replied:
Just an example, yes.

timoxley commented on app/stories/shared.stories.jsx:
```diff
@@ -262,3 +269,74 @@ story('Date Picker')
+        this.state = {
+            items: Array(5).fill(true).map((v, i) => `Item #${i}${i === 0 ? ' (Drag me!)' : ''}`),
+        }
+    }
```
Remember you can do away with `constructor` entirely:
```js
class SortableListContainer extends React.Component {
    state = {
        items: Array(5).fill(true).map((v, i) => `Item #${i}${i === 0 ? ' (Drag me!)' : ''}`),
    }
    …
}
```

This is the same syntax feature (class field declarations) you're already using for `onSortEnd = ({ newIndex, oldIndex }) => {`

timoxley commented on app/src/shared/components/FieldList/FieldItem/index.jsx:
```diff
@@ -0,0 +1,24 @@
+    </div>
+)
+
+FieldItem.styles = styles
```
👍

mondoreale commented on app/src/shared/components/SortableList/index.jsx:
```diff
@@ -0,0 +1,27 @@
+const SortableList = ({ children, ...props }: Props) => (
+    <div {...props}>
+        {React.Children.map(children, (child, index) => (
+            <SortableItem key={`item-${index}`} index={index}>
```
Actually… `key` comes from my previous internal iteration where I mapped an array to `SortableItem` instances (where `key` is required). It isn't for `React.Children.map`. Remove it.

mondoreale commented:
@timoxley thanks for the feedback! 🤠 

I rebased on `development`. Fixed few issues during a rebase (see 0abfc6e6d60c678d79ec6e6efc7c02f8825616dc). The rest got addressed in 
1e9cca3...bec85fd. Cheers!
